source/Patches/EndGameSummary.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using UnityEngine;
+using TownOfUs.Roles;
+using TownOfUs.Roles.Modifiers;
+using HarmonyLib;
+
+namespace TownOfUs.Patches
+{
+    [HarmonyPatch]
+    static class EndGameSummary
+    {
+        static class AdditionalTempData
+        {
+            public static List<PlayerInfo> playerData = new List<PlayerInfo>();
+
+            //leaving this in as i assume it maybe needed in future, feel free to remove if you think otherwise.
+            public static void clear()
+            {
+                playerData.Clear();
+            }
+
+            internal class PlayerInfo
+            {
+                public byte PlayerId { get; set; }
+                public string PlayerName { get; set; }
+                public Faction Faction { get; set; }
+                public Role CurrentRole { get; set; }
+                public Modifier Modifier { get; set; }
+                public int TasksCompleted { get; set; }
+                public int TasksTotal { get; set; }
+            }
+        }
+
+        public static void UpdatePlayerInfo()
+        {
+            Il2CppSystem.Collections.Generic.List<GameData.PlayerInfo> players = GameData.Instance.AllPlayers;
```

Comment:
Nit: I think we'd want to prefer the built-in C# List.

Reply:
i tried to use the built in list, but it did not work, an error of cannot implicitly convert between the two types.

I also tried casting and that did not seem to work either, so i stuck to what was working, as ugly as it looks, i used the normal List<T> where possible.

Reply:
> It was obtained from GameData.Instance.AllPlayers which is IL2CPP collections' List so that might require `ToArray()` (convert and cast it to built-in array type) and then `ToList()` (to built-in List collection).
> 
> This is unnecessary as it only was used once in the foreach loop. I'd recommend implicitly type it using `var` or directly reference it in the foreach loop (`foreach (GameData.PlayerInfo player in GameData.Instance.AllPlayers`).

I was using toArray().toList() but it threw the same error as that .toList() is a system list and it didnt like the conversion.

It could be there is a way this works, but I tried a  lot of ways and nothing seemed to. I had this issue with both PlayerControl and GameData.Instance.AllPlayers